argocd/resource_argocd_project.go

```diff
@@ -66,18 +65,14 @@ func resourceArgoCDProjectCreate(ctx context.Context, d *schema.ResourceData, me
 		}
 	}
 
-	c := *server.ProjectClient
 	projectName := objectMeta.Name
 
-	if _, ok := tokenMutexProjectMap[projectName]; !ok {
-		tokenMutexProjectMap[projectName] = &sync.RWMutex{}
-	}
+	obtainProjectLock(projectName, true)
```

Comment:
good idea, but I would make so that the obtainProjectLock function returns an unlock function to be called through the defer statement rather to mix both global variables and setters.

Reply:
I like that. Done.